sim-lib/src/lib.rs

```diff
@@ -770,7 +781,7 @@ impl Simulation {
             }
         });
 
-        let result_logger = Arc::new(Mutex::new(PaymentResultLogger::new()));
+        let result_logger = self.results.clone();
```

Comment:
Do we need both `let result_logger = self.results.clone();` and `let result_logger_clone = result_logger.clone();` below? Seems like we could reduce to a single clone?

Reply:
Hm. I think you need a results logger in the `run_results_logger` spawned task and the `consume_simulation_results` spawned task as well as the main simulation thread. So the Arc should have a reference count of 3? Which means you need to clone it twice. Am I understanding that correctly? Very possible I am missing something.

Reply:
Nope you're totally right - missed `consume_simulation_results` below!